src/tutorial/hello-world/README.md

```diff
@@ -1,2 +1,245 @@
+## Hello, world?: Huff
+
+### Getting Started
+
+Before you get started writing huff you will have to install the compiler. Head over to getting started and follow the steps to get it installed.
+Once that is done - come back here!! 
+
+### What you are going to learn?
+Unlike other programming languages, creating a huff contract that returns "hello world" is quite advanced! To keep things simple we are first going to learn how to create a huff contract that adds two numbers (then we will dive into "Hello, world"!).
+Open up your editor and create a file called `addTwo.huff`. Lets jump in. 
+
+### The Anatomy of a Huff Contract
+#### ABI declaration
+First things first. If you're coming from a higher level language like solidity or vyper you will be familiar with defining "external" or "public" functions to allow you to interact with your contract with external tools by generating an ABI (Application Binary Interface). This tells external tools or software languages how to target specific functions within a contract (We will dive more into this later). Huff is exactly the same, you can declare functions that will be appear in the abi at the top of the file. 
+
+```huff
+#define function addTwo(uint256, uint256) view returns(uint256)
+```
+
+Go ahead and paste the above example at the top of `addTwo.huff`. This means that we are declaring a function that takes two `uint256` inputs and returns a single `uint256`.
+
+#### The Main Macro
+The next thing we are going to create is the `main macro`. This serves as the entry point for our contract. All calls to your deployed contract (regardless of what function they are calling) will start from here! In this example we will define a MAIN function that will read two `uint256`'s from calldata and return their result.
+
+```huff
+#define macro MAIN() = takes(0) returns(0) {
+    0x00 calldataload     // [number1] // load first 32 bytes onto the stack - number 1     
+    0x20 calldataload     // [number2] // load second 32 bytes onto the stack - number 2
+    add                   // [number1+number2] // add number 1 and 2 and put the result onto the stack
+    
+    0x00 mstore           // place [number1 + number2] in memory
+    0x20 0x00 return      // return the result
+}
+```
+
+Looking at the above snippet may be intimidating at first, but bear with us.
+
+You'll notice that the MAIN directive is annotated with `takes(0) returns(0)`. As the EVM is a stack based virtual machine [TODO](see more), all macro declarations are annotated with the amount of stack items they will consume (take) and what they will add to the stack afterwards (returns). As this is the entry point to the program, and the stack is empty and we will not be leaving anything on the stack after execution, therefore takes and returns both have values of 0.
+
+Go ahead and copy the above macro into your `addTwo.huff` file. Run `huffc addTwo.huff --bytecode`.
+
+Congratulations you've just compiled your first contract!
```

Comment:
imo, getting a foundry test working for this new fn is important for the tutorial.  how else are they gonna know if they did it correctly?  wyt?  @abigger87 @cheethas

Reply:
That's not a bad idea, although I think testing has its own section - other options are using the vscode extension or link to an evm codes sandbox to step through execution.

Reply:
hmm actually thats a pretty good point.  i could roll with that.

Reply:
as long as there's some way for them to verify they did it right